.github/workflows/release.yml

```diff
@@ -0,0 +1,214 @@
+        with:
+          application_id: ${{ secrets.RELEASE_BOT_ID }}
+          application_private_key: ${{ secrets.RELEASE_BOT_PRIVATE_KEY }}
+          organization: "Lilypad-Tech"
+
+      # Approve with github-actions bot
+      - name: Approve PR
+        id: approve
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh pr review --approve
+
+      # Auto-merge PR when status checks complete or merge immediately if no checks in progress
+      # The auto-merge flag requires branch protection rule overrides.
+      - name: Merge Release PR
+        env:
+          GH_TOKEN: ${{ steps.get-release-bot-token.outputs.token }}
+        run: gh pr merge --squash --auto || gh pr merge --squash
+
+  extend-notes:
+    if: needs.release.outputs.releases_created == 'true' && needs.release.outputs.prs_created == 'false'
+    name: Extend Release Notes
+    needs: release
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.release.outputs.tag_name }}
+
+      - name: Get Release Bot Token
+        id: get-release-bot-token
+        uses: peter-murray/workflow-application-token-action@v1
+        with:
+          application_id: ${{ secrets.RELEASE_BOT_ID }}
+          application_private_key: ${{ secrets.RELEASE_BOT_PRIVATE_KEY }}
+          organization: "Lilypad-Tech"
+
+      - name: Extend Release Notes
+        env:
+          GH_TOKEN: ${{ steps.get-release-bot-token.outputs.token }}
+        run: |
+          BODY=$(gh release view ${{ needs.release.outputs.tag_name }} --json body | jq -r '.body')
+          EXTENSION=$(cat .github/releases/release_notes.md)
+
+          # Build up notes from BODY and EXTENSION seperated by two newlines
+          # https://trstringer.com/github-actions-multiline-strings/#option-2---environment-variable
+          NOTES=$(cat << EOF
+          $BODY
+
+          $EXTENSION
+          EOF
+          )
+          echo "NOTES<<EOF" >> $GITHUB_ENV
+          echo "$NOTES" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+          gh release edit ${{ needs.release.outputs.tag_name }} --notes "$NOTES"
+
+  publish-binaries:
+    if: needs.release.outputs.releases_created == 'true' && needs.release.outputs.prs_created == 'false'
+    name: Build and Publish Binaries
+    needs: release
+    strategy:
+      matrix:
+        include:
+          - goos: linux
+            goarch: amd64
+            gpu: true
+            runner: ubuntu-latest
+          - goos: linux
+            goarch: arm64
+            gpu: true
+            runner: linux-arm64
+          - goos: darwin
+            goarch: amd64
+            gpu: false
+            runner: macos-13 # uses amd64
+          - goos: darwin
+            goarch: arm64
+            gpu: false
+            runner: macos-latest # uses M1
+    runs-on: ${{ matrix.runner }}
```

Comment:
Nice! 😍

Reply:
Are we not doing _linux non gpu_? Those would be for people running the binary in docker? (assuming most docker images use some flavor of linux)

Reply:
We do a CPU-only build for every OS and architecture. The `gpu` flag indicates whether we should _also_ do a GPU build.

Reply:
Thank you for clarifying, I misread the matrix